### ISSUE TITLE
Parameters of the model ?

### ISSUE THREAD

**Issue by mohannadrateb** [oswaldoludwig/Seq2seq-Chatbot-for-Keras#14]
Hi , i just had a question regarding the parameters of the model, did you put them imperially or did you use an optimization technique to choose the parameters of the model ?

Thanks !!

**Comment by oswaldoludwig** [comment]
Hi, this space is for issues, not questions, but I can answer you question: no, it wasn't arbitrary, I reserved a small fraction of the training set as a validation set to observe the cross-entropy during training (I also printed the validation answers to have a better feeling about performance). The variable n_test in line 34 of train_bot.py defines the amount of data reserved for this validation process, now it’s set to a small value because I have already found the best architecture of this model to use as chatbot (for other uses, such as translation or text summarization, it's better to re-evaluate the architecture).
